src/execution.rs

```diff
@@ -141,111 +143,30 @@ impl ExecutionError {
     }
 }
 
-struct ExecutionContext<'a, 'tree> {
+struct ExecutionContext<'a, 'g, 's, 'tree> {
```

Comment:
One last nit: Do you know if this works without the extra lifetime parameters?  i.e., using `'a` down below as the parameters to `Globals` and `ScopedVariables`.  Often the compiler will be able to infer a "containing" lifetime, and that will simplify the types here a bit.

Reply:
I tried that first, but could not get it to work everywhere. I'll keep it like this, but if you think it is possible, I'm open to pairing on this to figure it out.